api4/user.go

```diff
@@ -84,6 +85,67 @@ func getUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getUsers(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
This route seems like it does too many things. Have we thought about splitting it into multiple routes?
For example:
/
/in_team/{team_id}
/in_channel/{channel_id}
/not_in_channel/{channel_id}

That way you could also tell the permission by the route.